transcom/mymove
# B-18155-Update-SIT-service-item-addresss-on-shipment-address-update

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a871784)

## Summary

This work will update the destination SIT service item's final address when the prime changes the shipment address.

Of the two endpoints called out in the description of the story: createSITAddressUpdateRequest "Creates an update request for a SIT service item's final delivery address. A newly created update request is assigned the status 'REQUESTED' if the change in address is > 50 miles and automatically approved otherwise." I didn't touch that endpoint as I believe there is some ambiguity we need to discuss on the approval process for shipment address changes and service item address changes. There is also a future story to address the logic for changing the status of a service item, B-18033.

The other endpoint in the description was updateMTOShipment. Instead of this, I made changes to updateMTOShipmentAddress. When the prime changes the shipment address, the destination SIT service item's destination address id will also change and will match the shipment. There is no logic for the TOO approval process as part of this work.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

To test, log in to the office app with the prime simulator role. Change the address of the shipment and verify in the database that the shipment and the SIT service items all share the same final destination address ID.

To test with the prime API, use the updateMTOShipmentAddress endpoint and provide the shipment ID, address ID that is being updated, and a request body with the address information, such as
{
  "id": "c56a4180-65aa-42ec-a945-5fd21dec0538",
  "streetAddress1": "123 Main Ave",
  "streetAddress2": "Apartment 9000",
  "streetAddress3": "Montmârtre",
  "city": "Anytown",
  "state": "AL",
  "postalCode": "90210",
  "country": "USA"
}

you can comment out the etag validation in mto_shipment_address_updater.go if you need to. When you get a successful response, again verify in the database that the addresses for the shipment and the SIT destination service items match what you provided as the address information.
